srcpkgs/openssl/template

```diff
@@ -0,0 +1,72 @@
+# Template file for 'openssl'
+pkgname=openssl
+version=1.1.1f
+revision=1
+build_style=configure
+configure_script="./Configure"
+configure_args="--prefix=/usr --openssldir=/etc/ssl --libdir=lib shared
+ no-async no-comp no-idea no-mdc2 no-rc5 no-ec2m no-sm2 no-sm4 no-ssl2
+ no-ssl3 no-seed no-weak-ssl-ciphers -Wa,--noexecstack"
+make_cmd=make
+make_build_args='MAKEDEPPROG="$(CC)'
+make_check_target=test
+hostmakedepends="perl"
+short_desc="Toolkit for Secure Sockets Layer and Transport Layer Security"
+maintainer="John <johnz@posteo.net>"
+license="OpenSSL-License"
+homepage="https://www.openssl.org"
+distfiles="https://www.openssl.org/source/openssl-${version}.tar.gz"
+checksum=186c6bfe6ecfba7a5b48c47f8a1673d0f3b0e5ba2e25602dd23b629975da3f35
+provides="libressl-3.3_${revision}"
+replaces="libressl>=0"
+
+case $XBPS_TARGET_MACHINE in
+	x86_64*) configure_args+=" enable-ec_nistp_64_gcc_128 linux-x86_64";;
+	i686*) configure_args+=" linux-elf";;
+	aarch64*) configure_args+=" linux-aarch64";;
+	ppc64le*) configure_args+=" linux-ppc64le";;
+	ppc64*) configure_args+=" linux-ppc64 no-asm";;
+	ppc*) configure_args+=" linux-ppc";;
+	arm*) configure_args+=" linux-armv4";;
+	mips*) configure_args+=" linux-mips32 -mips32";;
+	*) broken="$XBPS_TARGET_MACHINE";;
+esac
+
+pre_configure() {
+	configure_args+=" ${CPPFLAGS} ${CFLAGS} ${LDFLAGS}"
+}
+
+pre_build() {
+	make ${makejobs} depend
+}
+
+libcrypto1.1_package() {
+	short_desc+=" - crypto library"
+	pkg_install() {
+		vmove "usr/lib/libcrypto.so.*"
+		vmove usr/lib/engines-1.1
+	}
+}
+
+libssl1.1_package() {
+	short_desc+=" - SSL/TLS library"
+	pkg_install() {
+		vmove "usr/lib/libssl.so.*"
+	}
+}
+
+
+openssl-devel_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+	provides="libressl-devel-${version}_${revision}"
+	replaces="libressl>=0"
```

Comment:
I don't agree with this, why should this pkg `provide/replace` libressl?

Reply:
Right now it should be more like "conflicts". I only have this in here for local testing. And this "provides" libressl for the same reason libressl did provide openssl over all these years.

Reply:
This provides/replaces dance is due to the `openssl` command, if we use alternatives there's no need to do this.

Reply:
The shared libs don't conflicts because the SONAMEs differ, so we should just add a conflict in -devel.

Reply:
Like I said in the past, we can keep both at the same time just in case.